lex.go

```diff
@@ -43,14 +43,63 @@ var tokChanCap = TokenChanCap // capacity of lexer token channel
 func SetTokenChanCap(size int) {
 	tokChanCap = size
 }
-func Lex(reader io.Reader) chan NgxToken {
+
+// LexScanner provides an interface for scanning tokens.
```

Comment:
Nit: `LexScanner` _is_ and interface, it doesn't provide one. Maybe something like:

```suggestion
// LexScanner is the interface used by Lex to tokenize statements or blocks that may not conform to 
// the standard NGINX grammar. 
```

You could also provide additional detail to explain further how this type should be used. [See `bufio.Scanner` as an example](https://cs.opensource.google/go/go/+/refs/tags/go1.22.3:src/bufio/scan.go;l=14).

Looking at this again, I don't think that `LexScanner` needs to be an interface at all as there is only a single implementation of the interface (`*extScanner`) and that could probably be used when necessary in tests.

Reply:
I removed `LexScanner` and use `*extScanner` instead